src/main/java/org/jenkinsci/remoting/protocol/impl/NIONetworkLayer.java

```diff
@@ -147,12 +147,12 @@ public void ready(boolean accept, boolean connect, boolean read, boolean write)
                             while (recv.hasRemaining()) {
                                 onRead(recv);
                             }
-                            if (in.isOpen() && recvKey.isValid()) {
-                                getIoHub().addInterestRead(recvKey);
-                            } else {
-                                recvKey.cancel();
-                                onRecvClosed();
-                            }
+                        }
+                        if (in.isOpen() && recvKey.isValid()) {
```

Comment:
I think because -1 means the channel is closed and the key has been cancelled already... but that is just a guess